### ISSUE TITLE
Rule Change: [id-denylist] allow configuring private identifiers separately

### ISSUE THREAD

**Issue by jtbandes** [eslint/eslint#17138]
### What rule do you want to change?

id-denylist

### What change to do you want to make?

Generate fewer warnings

### How do you think the change should be implemented?

A new option

### Example code

```js
class C {
  #window = {};
}
```

### What does the rule currently do for this code?

When configured with `id-denylist: ["error", "window"]`, the rule produces an error: `Identifier '#window' is restricted.`

### What will the rule do after it's changed?

It would be nice to allow configuring private and non-private identifiers separately. Since the private identifier `#window` does not risk conflicting with the global `window`, there's no reason to disallow it.

A new option or options could be provided to allow configuring the denylist separately for private and non-private identifiers.

### Participation

- [X] I am willing to submit a pull request to implement this change.

**Comment by Rec0iL99** [comment]
Hi @jtbandes, thanks for the issue. We currently aren't adding any new options to stylistic rules due to the [stylistic rule change policy](https://github.com/eslint/eslint#stylistic-rule-updates). You can always copy the core rule into a custom rule and add new options.

**Comment by jtbandes** [comment]
Thanks for the reply. The policy mentions “…unless an option is the only way to fix a bug or support a newly-added ECMAScript feature”. I believe this is a language feature that was not around when the rule was originally made, so I think it would be appropriate to add an option allowing the user to control the interaction between this rule and the new feature.

**Comment by Rec0iL99** [comment]
I'm sorry that I missed that. I believe this is the [proposal](https://github.com/tc39/proposal-class-fields).

**Comment by jtbandes** [comment]
See also: https://github.com/eslint/eslint/pull/14591 (the rule was upgraded to support private fields in this PR, but it seems like the update was mainly intended to avoid breakage rather than to add first-class support for private fields).

**Comment by mdjermanovic** [comment]
@nzakas what do you think about this? I'm not in favor of separating configurations for private and non-private identifiers for the reasons described in https://github.com/eslint/eslint/issues/17138#issuecomment-1545855547, but curious about your opinion.

**Comment by nzakas** [comment]
@mdjermanovic I agree with your reasoning. This rule is about disallowing certain names to be used *in any way*. We don't make distinctions between properties and variables, so I can't think of a reason to make a distinction about private names.

Of course, you can always fork the rule and make it do what you want, but I don't think it's a good fit for the core.
